In [2]:
import pandas as pd

In [3]:
sales = pd.read_csv("sales.csv")

In [4]:
%%sql
SELECT
    warehouse,
    SUM(total) AS total_sales
FROM
    sales
GROUP BY
    warehouse;

warehouse  total_sales
0     North    100203.63
1   Central    141982.88
2      West     46926.49

In [6]:
%%sql
WITH warehouse_sales AS (
    SELECT
        warehouse,
        SUM(total) AS total_sales
    FROM
        sales
    GROUP BY
        warehouse
)
SELECT
    warehouse,
    total_sales,
    CASE
        WHEN total_sales > (SELECT AVG(total_sales) FROM warehouse_sales) THEN 'Outperforms'
        WHEN total_sales = (SELECT MAX(total_sales) FROM warehouse_sales) THEN 'Ties for the top performer'
        ELSE 'Does not outperform'
    END AS performance
FROM
    warehouse_sales;

warehouse  total_sales          performance
0     North    100203.63          Outperforms
1   Central    141982.88          Outperforms
2      West     46926.49  Does not outperform

In [7]:
%%sql
SELECT
    strftime('%Y-%m', CAST(date AS DATE)) AS month,
    warehouse,
    SUM(total) AS total_sales
FROM
    sales
GROUP BY
    month, warehouse
ORDER BY
    month, warehouse;

month warehouse  total_sales
0  2021-06   Central     44128.96
1  2021-06     North     33318.43
2  2021-06      West     17872.64
3  2021-07   Central     48269.70
4  2021-07     North     29122.94
5  2021-07      West     16155.27
6  2021-08   Central     49584.22
7  2021-08     North     37762.26
8  2021-08      West     12898.58

# Warehouse Performance Analysis

This report provides an analysis of the sales performance across different warehouses (North, Central, West) based on the provided dataset.

## Overview

The warehouse performance analysis aims to understand how sales vary between different warehouses and identify any seasonal or regional trends affecting sales in specific locations.

## Findings

### Sales Trends by Warehouse

- **June 2021**:
  - Central warehouse had the highest sales ($44,128.96), followed by North ($33,318.43), and West ($17,872.64).
- **July 2021**:
  - Central warehouse led in sales again ($48,269.70), followed by North ($29,122.94), and West ($16,155.27).
- **August 2021**:
  - Central warehouse maintained its lead in sales ($49,584.22), followed by North ($37,762.26), and West ($12,898.58).

### Observations

1. **Seasonal Trends**:
   - Sales across all warehouses show consistency in performance from June to August, with Central consistently leading in sales followed by North, and then West.
   - There are no significant deviations in sales patterns between months, indicating a stable performance across the summer period.

2. **Regional Trends**:
   - Central warehouse consistently outperforms others in terms of sales volume, suggesting higher demand or better market conditions in that region.
   - North warehouse follows Central in sales performance, maintaining moderate sales volume throughout the period.
   - West warehouse consistently lags behind Central and North in sales, indicating potential regional factors impacting its performance.

## Conclusion

The analysis of warehouse performance provides insights into the sales dynamics across different locations. Central warehouse emerges as the top performer, consistently leading in sales, followed by North, while West lags behind. Understanding these regional trends can inform strategic decisions to optimize sales strategies and improve overall performance in specific regions.